crates/adapters/src/jit/mod.rs

```diff
@@ -219,18 +225,33 @@ pub fn start_circuit(
 
         // FIXME: This is unsafe. The correct fix is to make sure `endpoint.disconnect`
         // returns after all endpoint threads have terminated.
-        let json =
-            unsafe { circuit.serialization_function(json_output_demands[&node_id], layout_id) }
-                .ok_or_else(|| {
-                    ControllerError::jit_error(&format!(
-                "JSON serialization function not found (view name: '{}', layout id: {layout_id})",
+        let default_json = unsafe {
+            circuit.serialization_function(default_json_output_demands[&node_id], layout_id)
+        }
+        .ok_or_else(|| {
+            ControllerError::jit_error(&format!(
+                "Default JSON serialization function not found (view name: '{}', layout id: {layout_id})",
                 table_schema.name,
             ))
-                })?;
+        })?;
+
+        let snowflake_json = unsafe {
```

Comment:
so you generate code for all ser/deser variants and dynamically choose one?

Reply:
for now - yes.  The JIT compiler is capable of generating (de)serializers at runtime, we should be able to do this in the future.